### ISSUE TITLE
cljr-rename-symbol doesn't rename partials

### ISSUE THREAD

**Issue by JAremko** [clojure-emacs/clj-refactor.el#429]
## Steps to reproduce the problem

Run `cljr-rename-symbol` with `foo` renamed to `bar` on 

``` clojure
(defn foo [arg1 arg2])
(bar 1 2)
(partial foo 1)
```

## Expected behavior

``` clojure
(defn bar [arg1 arg2])
(bar 1 2)
(partial bar 1)
```

## Actual behavior

``` clojure
(defn bar [arg1 arg2])
(bar 1 2)
(partial foo 1)
```

### CIDER version
```
;; CIDER 0.18.0snapshot (package: 20180802.1919), nREPL 0.2.13
;; Clojure 1.9.0, Java 10.0.1
```

### Lein version
Leiningen 2.8.1 on Java 10.0.1 OpenJDK 64-Bit Server VM

### Emacs version
26.1

**Comment by expez** [comment]
Can't reproduce.  From `lein new test` I got:

```clj
(ns test.core)

(defn foo
  "I don't do a whole lot."
  [x]
  (println x "Hello, World!"))

(foo 1)

(partial foo 1)
```

Rename `foo` to `bar`:

```clj
(ns test.core)

(defn bar
  "I don't do a whole lot."
  [x]
  (println x "Hello, World!"))

(bar 1)

(partial bar 1)
```

You already had a bar and wanted foo to have the same name or something?  Need more details to reproduce.

**Comment by JAremko** [comment]
@expez  bad copy-pasta in the first listing. fixed.

**Comment by JAremko** [comment]
This is what I'm getting 
![https://i.imgur.com/oXPZD7z.png](https://i.imgur.com/oXPZD7z.png)
![https://i.imgur.com/xBNvQhC.png](https://i.imgur.com/xBNvQhC.png)

**Comment by expez** [comment]
Do you get the same if you load the code into the REPL before doing the refactoring?

E.g. by using <kbd>C-c C-k</kbd> in CIDER?

**Comment by benedekfazekas** [comment]
i can see that you are using prettifing symbols on partial. try the feature
with prettifing switched off. it might confuse the column counts cljr works
with.

**Comment by JAremko** [comment]
So basically `cljr-rename-symbol` conflicts with  [clojure-enable-fancify-symbols](https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Blang/clojure/config.el#L20) in Spacemacs.  I'm just not sure how [font lock shenanigans](https://github.com/syl20bnr/spacemacs/blob/develop/layers/+lang/clojure/funcs.el#L12) can affect `cljr-rename-symbol`.

**Comment by JAremko** [comment]
> it might confuse the column counts cljr works

@benedekfazekas   Just caked.  You are right. It affects column number.

**Comment by benedekfazekas** [comment]
yeah. used to love prettify symbols too. but it messes with cljr and even
more importantly with indentation. I still use it but very tuned down.

**Comment by expez** [comment]
Is there a way to get the 'real' column number?  If not, then I don't see how cljr could handle this...

**Comment by JAremko** [comment]
@expez Indirect buffers probably can be used for this. They can have different modes then the original one.

**Comment by JAremko** [comment]
@expez  Also in this case  `composite.el` has function [`find-composition`](https://github.com/emacs-mirror/emacs/blob/master/lisp/composite.el#L302) that probably can help.

**Comment by JAremko** [comment]
Keep in mind that this also doesn't work:

``` clojure
(defn foo
  "I don't do a whole lot."
  [x]
  (println x "Hello, World!"))

(foo 1)

(partial map identity) (foo 1)
```

I don't now how `refactor.el` actually does substitution but from what i gather the best way to solve it will be `letf`-ing prettyfication list for clojure mode around the substitution logic to empty one - so nothing will be "prettyfied" when you visit files for substitution.  To deal with files that already have visiting buffer you can create `Indirect buffers` of those without `pretty-mode` and substitute in them.

**Comment by expez** [comment]
> no way that will know about prettify symbols

Thankfully it doesn't have to, because the pretty symbols only really exist in emacs.  The regular text is what's stored in the file on disk.

**Comment by benedekfazekas** [comment]
right. sorry, what i meant is that the line/cols returned by the middleware
are not emacs buffer related

**Comment by Nimamoh** [comment]
has anyone found a workaround for this issue?

**Comment by JAremko** [comment]
@Nimamoh  I ended up simply disabling fancy symbols primary because they also mess up long line  highlight.
